src/message/__tests__/messageActionSheet-test.js

```diff
@@ -0,0 +1,115 @@
+import { constructActionButtons } from '../messageActionSheet';
+import {
+  streamNarrow,
+  homeNarrow,
+  specialNarrow,
+  privateNarrow,
+  topicNarrow,
+  groupNarrow,
+} from '../../utils/narrow';
+
+describe('constructActionButtons', () => {
+  const auth = {
+    realm: '',
+    email: 'Zoe@zulip.com',
+  };
+
+  let message = {};
+  let subscriptions = [];
+  let mute = [];
+  let narrow = [];
+
+  test('show narrow option in stream narrow', () => {
+    narrow = streamNarrow('all');
+    const buttons = constructActionButtons({ message, auth, narrow, subscriptions, mute });
+
+    expect(buttons).toContain('Narrow to conversation');
+  });
+
+  test('show narrow option in home narrow', () => {
+    narrow = homeNarrow();
+    const buttons = constructActionButtons({ message, auth, narrow, subscriptions, mute });
+
+    expect(buttons).toContain('Narrow to conversation');
+  });
+
+  test('show narrow option in special narrow', () => {
+    narrow = specialNarrow('private');
+    const buttons = constructActionButtons({ message, auth, narrow, subscriptions, mute });
+
+    expect(buttons).toContain('Narrow to conversation');
+  });
+
+  test('do not show narrow option in private narrow', () => {
+    narrow = privateNarrow('abc@zulip.com');
+    const buttons = constructActionButtons({ message, auth, narrow, subscriptions, mute });
+
+    expect(buttons).not.toContain('Narrow to conversation');
```

Comment:
I think the 'Narrow to conversation' tests are a bit too repetitive. Can you combine them into one?